tests/ccl_test_emu.c

```diff
@@ -0,0 +1,192 @@
+    if(status!=7) {
+      fprintf(stderr,"Error reading file %s, line %d\n",fname,i);
+      exit(1);
+    }
+    data->w_0[i] = w_0[i];
+    data->w_a[i] = w_a[i];
+    data->h[i] = h[i];
+    data->sigma8[i] = sigma8[i];
+    data->Omega_c[i] = Omega_c[i];
+    data->Omega_b[i] = Omega_b[i];
+    data->n_s[i] = n_s[i];
+  }
+  fclose(f);
+}
+
+static int linecount(FILE *f)
+{
+  //////
+  // Counts #lines from file
+  int i0=0;
+  char ch[1000];
+  while((fgets(ch,sizeof(ch),f))!=NULL) {
+    i0++;
+  }
+  return i0;
+}
+
+static void compare_emu(int i_model,struct emu_data * data)
+{
+  int nk,i,j;
+  int status=0;
+  char fname[256],str[1024];
+  char* rtn;
+  FILE *f;
+  int i_model_vec[6]={1,3,5,6,8,10};
+  //The emulator cosmologies we can compare to
+  //without CLASS failing due to w(z) crossing -1.
+  
+  ccl_configuration config = default_config;
+  config.transfer_function_method = ccl_emulator;
+  config.matter_power_spectrum_method = ccl_emu;
+  
+  //None of the current cosmologies being checked include neutrinos
+  ccl_parameters params = ccl_parameters_create(data->Omega_c[i_model-1],data->Omega_b[i_model-1],0.0,data->Neff, data->mnu, data->mnu_type, data->w_0[i_model-1],data->w_a[i_model-1],data->h[i_model-1],data->sigma8[i_model-1],data->n_s[i_model-1],-1,-1,-1,-1,NULL,NULL, &status);
+  params.Omega_g=0;
+  params.sigma8=data->sigma8[i_model-1];
+  ccl_cosmology * cosmo = ccl_cosmology_create(params, config);
+  ASSERT_NOT_NULL(cosmo);
+
+  //Each of these files has the smoothed simulated power spectrum for
+  //the corresponding cosmology, kindly provided by E. Lawrence.
+  sprintf(fname,"./tests/benchmark/emu_smooth_pk_M%d.txt",i_model_vec[i_model-1]);
+  f=fopen(fname,"r");
+  if(f==NULL) {
+    fprintf(stderr,"Error opening file %s\n",fname);
+    exit(1);
+  }
+  nk=linecount(f)-1; rewind(f);
+  
+  double k=0.,pk_bench=0.,pk_ccl,err;
+  double z=0.; //Other redshift checks are possible but not currently implemented
+  int stat=0;
+  
+  for(i=0;i<nk;i++) {      
+    stat=fscanf(f,"%le %le\n",&k, &pk_bench);
+    if(stat!=2) {
+      fprintf(stderr,"Error reading file %s, line %d\n",fname,i);
+      exit(1);
+    }
+    pk_ccl=ccl_nonlin_matter_power(cosmo,k,1./(1+z),&status);
+    if (status) printf("%s\n",cosmo->status_message);
+    err=fabs(pk_ccl/pk_bench-1);
+    ASSERT_DBL_NEAR_TOL(err,0.,EMU_TOLERANCE);
+  }
+
+  fclose(f);
+
+  ccl_cosmology_free(cosmo);
+}
+
+//Cosmology M001
+CTEST2(emu,model_1) {
+  int model=1;
+  compare_emu(model,data);
+}
+
+/*
+//Additional tests for other cosmologies are possible:
+CTEST2(emu,model_2) {
+  int model=2;
+  compare_emu(model,data);
+}
+
+CTEST2(emu,model_3) {
+  int model=3;
+  compare_emu(model,data);
+}
+
+CTEST2(emu,model_4) {
+  int model=4;
+  compare_emu(model,data);
+}
+
+CTEST2(emu,model_5) {
+  int model=5;
+  compare_emu(model,data);
+}
+
+CTEST2(emu,model_6) {
+  int model=6;
+  compare_emu(model,data);
+}
```

Comment:
Why aren't these implemented?  It looks like the benchmark files are all there.  

If they are just slow, so you don't want to run them on travis, you can have them inside a `#ifndef SKIP_SLOW_TESTS` guard, and have the Travis compilation add `-DSKIP_SLOW_TESTS` so they don't get run there.  But manual running of tests (which you should do periodically, especially before new version releases) would run them.

Or you could do it the other way with `#ifdef RUN_SLOW_TESTS` and have a special pre-release setup that you do that sets `-DRUN_SLOW_TESTS` so you can run these.

Either way, but it seems like these tests are probably worthwhile and should be easy to enable for at least occasional testing.

Reply:
Indeed this was for speed. We'll implement this.

Reply:
We have compromised in the following solution now: enabling all tests that were commented out and opening an issue to add these flags later one.